File: PDOO/Napakalaki/src/NapakalakiGame/CultistPlayer.java
/*
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */
package NapakalakiGame;

import java.util.Random;

public class CultistPlayer extends Player {

    private static int totalCultistPlayers = 0;
    private Cultist myCultistCard = null;

    public CultistPlayer(Player p, Cultist c) {
        super(p.getName());
        this.copyPlayer(p);
        myCultistCard = c;
        totalCultistPlayers++;
    }

    public static int getTotalCultistPlayers() {
        return totalCultistPlayers;
    }

    public int getCombatLevel() {
        double valor = super.getCombatLevel();
        valor += valor * 0.2;
        valor += (myCultistCard.getGainedLevels() * CultistPlayer.getTotalCultistPlayers());
        return (int) valor;
    }

    protected int getOponentLevel(Monster m) {
        return m.getCombatLevelAgainstCultistPlayer();
    }

    protected boolean shouldConvert() {
        return false;
    }

    private Treasure giveMeATreasure() {
        if (canYouGiveMeATreasure()) {
            Random r = new Random();
            int indice = r.nextInt(this.getVisibleTreasures().size()) + 1;
            Treasure tesoro = this.getVisibleTreasures().get(indice);
            getVisibleTreasures().remove(indice);
            return tesoro;
        } else {
            return null;
        }
    }

    private boolean canYouGiveMeATreasure() {
        if (getVisibleTreasures().size() > 0) {
            return true;
        } else {
            return false;
        }
    }

}

File: PDOO/Napakalaki-Ruby/lib/cultist_player.rb
# To change this license header, choose License Headers in Project Properties.
# To change this template file, choose Tools | Templates
# and open the template in the editor.

module NapakalakiGame
  require_relative "cultist.rb"
  
  class CultistPlayer < Player
    @@totalCultistPlayers = 0
    def initialize(p, c)
      super(p.getName)
      self.copyPlayer(p)
      @myCultistCard = c
      @@totalCultistPlayers = @@totalCultistPlayers + 1    
    end
  
    protected
  
    def getCombatLevel
      valor = super
      valor += valor * 0.2
      valor += (@myCultistCard.getGainedLevels * CultistPlayer.getTotalCultistPlayers)
      valor = valor.truncate
      return valor
    end
  
    def getOponentLevel(m)
      return m.getCombatLevelAgainstCultistPlayer
    end
  
    def shouldConvert
      return false
    end
  
    private
  
    def giveMeATreasure
      if(canYouGiveMeATreasure)
        indice=rand(getVisibleTreasures.size)+1
        tesoro=getVisibleTreasures.at(indice)
        getVisibleTreasures.delete_at(indice)
        return tesoro
      else
        return nil
      end
    end
  
    def canYouGiveMeATreasure
      if(getVisibleTreasures.size > 0)
        puede = true
      else
        puede = false
      end
      return puede
    end
  
    public
  
    def self.getTotalCultistPlayers
      return @@totalCultistPlayers
    end
  end
end
